### PR TITLE
feat: Salesforce connector accepts key path or value (#2321)

### PR DESCRIPTION
**Pull request by jakub-sandomierz-deepsense-ai**
Solution to issue https://github.com/Unstructured-IO/unstructured/issues/2321.

simple_salesforce API allows for passing private key path or value. This PR introduces this support for Ingest connector.

Salesforce parameter "private-key-file" has been renamed to "private-key".
It can contain one of following:
- path to PEM encoded key file (as string)
- key contents (PEM encoded string)

If the provided value cannot be parsed as PEM encoded private key, then the file existence is checked. This way private key contents are not exposed to unnecessary underlying function calls.

### CODE REVIEW

File: test_unstructured_ingest/unit/test_salesforce_connector.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+def pkey_to_str(key) -> str:
+    return key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.PKCS8,
+        encryption_algorithm=serialization.NoEncryption(),
+    ).decode("utf-8")
+
+
+def rsa_private_key() -> str:
+    return pkey_to_str(rsa.generate_private_key(0x10001, 512))
```
[the remark this answers is not in the document]
Comment (jakub-sandomierz-deepsense-ai) [reply test_unstructured_ingest/unit/test_salesforce_connector.py:20]: Keys are generated for validation purposes and are as short as possible. They are not used in any connection.

File: test_unstructured_ingest/unit/test_salesforce_connector.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def brainpoolp512r1_private_key() -> str:
+    return pkey_to_str(ec.generate_private_key(ec.BrainpoolP512R1))
+
+
+def dsa_private_key() -> str:
+    return pkey_to_str(dsa.generate_private_key(1024))
```
[the remark this answers is not in the document]
Comment (jakub-sandomierz-deepsense-ai) [reply test_unstructured_ingest/unit/test_salesforce_connector.py:28]: Keys are generated for validation purposes and are as short as possible. They are not used in any connection.

File: unstructured/ingest/cli/cmds/salesforce.py [hunk 3]
```diff
@@ -31,11 +32,11 @@ def get_cli_options() -> t.List[click.Option]:
                 help="For the Salesforce JWT auth. Found in Consumer Details.",
             ),
             click.Option(
-                ["--private-key-path"],
+                ["--private-key"],
                 required=True,
-                type=click.Path(file_okay=True, exists=True, dir_okay=False),
-                help="Path to the private key for the Salesforce JWT auth. "
-                "Usually named server.key.",
+                type=FileOrJson(allow_raw_str=True),
```
Comment (rbiseck3) [inline unstructured/ingest/cli/cmds/salesforce.py:37]: Will the contents of the private key always be a valid json? If so, lets use the default `allow_raw_str` to raise an error if the value passed in is not a valid json.

**Comment by rbiseck3** [comment]
Will need to bump the version on the changelog as well as the `__version__.py` file

File: unstructured/ingest/cli/cmds/salesforce.py (hunk 3 above)
Comment (jakub-sandomierz-deepsense-ai) [reply unstructured/ingest/cli/cmds/salesforce.py]: I have changed type to `str`. This parameter should hold path or PEM key. Full validation is done in `SalesforceAccessConfig`.

**Comment by jakub-sandomierz-deepsense-ai** [comment]
> Will need to bump the version on the changelog as well as the `__version__.py` file

I have merged main and moved my changes in changelog to current version. `__version__.py` contains current version too.

**Review by rbiseck3 — APPROVED** [review]

**Pull request merged**